### ISSUE TITLE
[Bug][TGL][SDW] Can't hear audio output when aplay via headset in first time.

### ISSUE THREAD

**Issue by ClarexZhou** [thesofproject/linux#1333]
**Describe the bug**
Can't heart audio output when aplay via headset in first time. And can hear audio output normally if release first one and trigger again.

Might i915 related. Error only reproduced if unit system boot up without eDP display. Error can't be reproduced if system boot up with eDP display connected.

**To Reproduce**
1. Headset is connected 
2. Do system reboot via "sudo reboot"
3. aplay -Dhw:0,0 -c 2 -r 48000 -f s16_le ~/wav/48California_Dreaming.wav -vvv -i 

Headset isn't detected after step2:
```
numid=10,iface=CARD,name='Headphone Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=off
numid=11,iface=CARD,name='Headset Mic Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=off
```
Headset can be detected after step3:
```
numid=10,iface=CARD,name='Headphone Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=on
numid=11,iface=CARD,name='Headset Mic Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=on
```

**frequency**
100%

**Actual result**
No audio output on headset. 

More info:
Close this aplay and try aplay again, audio can output normally.

**Expected result**
Audio should output in first time

**Environment**
Kernel: 0ebe10b on integration/soundwire-latest
Firmware: https://github.com/thesofproject/sof/tree/tgl-002-drop-stable  tip commit c8bc4b2. 
Topology: file: Master SOF branch commit: b5982a4 + PR1939
platform: SDW TGL RVP with onboard ALC711 + ALC1308
[amixer_after1stAplay.txt](https://github.com/thesofproject/linux/files/3738521/amixer_after1stAplay.txt)
[amixer_beforeAplay.txt](https://github.com/thesofproject/linux/files/3738522/amixer_beforeAplay.txt)
[dmesg_after1stAplay.txt](https://github.com/thesofproject/linux/files/3738523/dmesg_after1stAplay.txt)

**Comment by ClarexZhou** [comment]
Once this issue occurred, arecord via headset always can't capture any audio. 
```
ubuntu@TGL-U-I2S:~$ arecord -Dhw:0,1 -r48000 -c2 -fs16_le test.wav -vvv -i
Recording WAVE 'test.wav' : Signed 16 bit Little Endian, Rate 48000 Hz, Stereo
Hardware PCM card 0 'sof-rt711-rt1308' device 1 subdevice 0
Its setup is:
  stream       : CAPTURE
  access       : RW_INTERLEAVED
  format       : S16_LE
  subformat    : STD
  channels     : 2
  rate         : 48000
  exact rate   : 48000 (48000/1)
  msbits       : 16
  buffer_size  : 16368
  period_size  : 4080
  period_time  : 85000
  tstamp_mode  : NONE
  tstamp_type  : MONOTONIC
  period_step  : 1
  avail_min    : 4080
  period_event : 0
  start_threshold  : 1
  stop_threshold   : 16368
  silence_threshold: 0
  silence_size : 0
  boundary     : 9214364837600034816
  appl_ptr     : 0
  hw_ptr       : 0
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
Max peak (8160 samples): 0x00000000 #                    0%
```
[dmesg_arecord.txt](https://github.com/thesofproject/linux/files/3738615/dmesg_arecord.txt)

**Comment by dengyangchao** [comment]
observed one change
when aplay via headset in the first times it can heard the output audio now, and other issue still exist

Test recipe:
kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 7c85d69
FW: https://github.com/thesofproject/sof/commits/master commit: 8b059f6
tplg: Same with TGL FW branch commit: sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711-rt1308.tplg
platform: TGL RVP, with codec ACL711, ALC1308 in SDW mode

[dmesg.log](https://github.com/thesofproject/linux/files/3760947/dmesg.log)
[logger.log](https://github.com/thesofproject/linux/files/3760948/logger.log)

**Comment by bardliao** [comment]
> observed one change
> when aplay via headset in the first times it can heard the output audio now, and other issue still exist
> 
> Test recipe:
> kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 7c85d69
> FW: https://github.com/thesofproject/sof/commits/master commit: [8b059f6](https://github.com/thesofproject/linux/commit/8b059f62ad2cbcccb7be0d889e01018c1e2d699b)
> tplg: Same with TGL FW branch commit: sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711-rt1308.tplg
> platform: TGL RVP, with codec ACL711, ALC1308 in SDW mode
> 
> [dmesg.log](https://github.com/thesofproject/linux/files/3760947/dmesg.log)
> [logger.log](https://github.com/thesofproject/linux/files/3760948/logger.log)

@dengyangchao What did you mean by other issues? Did you mean 'Headphone Jack' and 'Headset Mic Jack' didn't detected and no audio can be captured? Could you blacklist snd_sof_pci and run ./sof_insert.sh after boot?

**Comment by dengyangchao** [comment]
with the latest recipe below, in volume controler, the headphone and headset mic is muted in default setting. when open it, can heard the audio output via headset.
![alsamixer](https://user-images.githubusercontent.com/52945407/67547874-cfe7a100-f732-11e9-8684-0b98a501df4f.PNG)
exist:
1.headset aplay(0,0) output normally if unmute the headphone in alsamixer
2.always can't detect the headset
```
 numid=10,iface=CARD,name='Headphone Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=off
numid=11,iface=CARD,name='Headset Mic Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=off
```
3. arecord via headset(0,1) can't capture voice

**Tried**
1. connected the eDP display,  issue 2 and 3 still exist

2. blacklist the snd_sof_pci ,  issue 2 and 3 still exist

kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest  commit:  1c3de3d 
sof: https://github.com/thesofproject/sof/commits/master   commit: 89ef0af
Tplg: Same with TGL FW branch commit:  sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711-rt1308.tplg


<details>
 <summary>logger-error</summary>

````
 CORE  LEVEL      COMP_ID                TIMESTAMP            DELTA                FILE_NAME    CONTENT
    0      1           SA         267581780.208333 267581776.000000      src/lib/agent.c:57     validate(), ll drift detected, delta = 40463
    0      1          ALH         309749147.031250  42167368.000000 ers/intel/cavs/alh.c:70     alh_get_handshake() error: stream_id 65535 out of range

````
</details>


[dmesg.log](https://github.com/thesofproject/linux/files/3770667/dmesg.log)
[headset_logger_error.log](https://github.com/thesofproject/linux/files/3770668/headset_logger_error.log)
[logger.log](https://github.com/thesofproject/linux/files/3770669/logger.log)

**Comment by bardliao** [comment]
@dengyangchao Can you change rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c and test it again?
The code is at [rt711->jd_src = RT711_JD2;](https://github.com/thesofproject/linux/blob/integration/soundwire-latest/sound/soc/codecs/rt711.c#L1155)
The HW design between RVPs and laptops are different. RVPs uses JD1, but laptop uses JD2.

**Comment by dengyangchao** [comment]
> @dengyangchao Can you change rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c and test it again?
> The code is at [rt711->jd_src = RT711_JD2;](https://github.com/thesofproject/linux/blob/integration/soundwire-latest/sound/soc/codecs/rt711.c#L1155)
> The HW design between RVPs and laptops are different. RVPs uses JD1, but laptop uses JD2.

after change JD2 -> JD1.
the first time still can't heard the audio output via headset(0,0) and can't detect headset
if trigger again it can heard the audio output and detected headset. 
and headset arecord(0,1) still can't capture any voice

add snd_sof_pci to blacklist,
1. boot up and aplay -l shows "no soundcards",
2. then run ./sof_insert.sh, and aplay -l still show "no soundcard"
3. then run ./sof_remove.sh  and ./sof_insert.sh, aplay -l can see the pcm list,
4. "amixer contents" shows detected the headset and aplay headset can heard the audio output
5. headset arecord(0,1) can capture voice normally

kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 1c3de3d
sof: https://github.com/thesofproject/sof/commits/master commit: 89ef0af
Tplg: Same with TGL FW branch commit: sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711-rt1308.tplg

[dmesg_blacklist.log](https://github.com/thesofproject/linux/files/3771028/dmesg_blacklist.log)
[logger.log](https://github.com/thesofproject/linux/files/3771030/logger.log)

**Comment by dengyangchao** [comment]
> @dengyangchao Can you change rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c and test it again?
> The code is at [rt711->jd_src = RT711_JD2;](https://github.com/thesofproject/linux/blob/integration/soundwire-latest/sound/soc/codecs/rt711.c#L1155)
> The HW design between RVPs and laptops are different. RVPs uses JD1, but laptop uses JD2.

also tried if we don't add snd_sof_pci to blacklist.
when boot up and run ./sof_remove.sh and ./sof_insert.sh.  it can detect the headset, and both headset playback and capture are work normally

**Comment by bardliao** [comment]
> > @dengyangchao Can you change rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c and test it again?
> > The code is at [rt711->jd_src = RT711_JD2;](https://github.com/thesofproject/linux/blob/integration/soundwire-latest/sound/soc/codecs/rt711.c#L1155)
> > The HW design between RVPs and laptops are different. RVPs uses JD1, but laptop uses JD2.
> 
> after change JD2 -> JD1.
> the first time still can't heard the audio output via headset(0,0) and can't detect headset
> if trigger again it can heard the audio output and detected headset.
> and headset arecord(0,1) still can't capture any voice
> 
> add snd_sof_pci to blacklist,
> 
> 1. boot up and aplay -l shows "no soundcards",
> 2. then run ./sof_insert.sh, and aplay -l still show "no soundcard"
> 3. then run ./sof_remove.sh  and ./sof_insert.sh, aplay -l can see the pcm list,
> 4. "amixer contents" shows detected the headset and aplay headset can heard the audio output
> 5. headset arecord(0,1) can capture voice normally
> 

@dengyangchao Could you share the dmesg log after doing step 1 and 2?
I would like to know why aplay -l still show "no soundcard" after  ./sof_insert.sh.

Thanks

**Comment by dengyangchao** [comment]
> > > @dengyangchao Can you change rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c and test it again?
> > > The code is at [rt711->jd_src = RT711_JD2;](https://github.com/thesofproject/linux/blob/integration/soundwire-latest/sound/soc/codecs/rt711.c#L1155)
> > > The HW design between RVPs and laptops are different. RVPs uses JD1, but laptop uses JD2.
> > 
> > 
> > after change JD2 -> JD1.
> > the first time still can't heard the audio output via headset(0,0) and can't detect headset
> > if trigger again it can heard the audio output and detected headset.
> > and headset arecord(0,1) still can't capture any voice
> > add snd_sof_pci to blacklist,
> > 
> > 1. boot up and aplay -l shows "no soundcards",
> > 2. then run ./sof_insert.sh, and aplay -l still show "no soundcard"
> > 3. then run ./sof_remove.sh  and ./sof_insert.sh, aplay -l can see the pcm list,
> > 4. "amixer contents" shows detected the headset and aplay headset can heard the audio output
> > 5. headset arecord(0,1) can capture voice normally
> 
> @dengyangchao Could you share the dmesg log after doing step 1 and 2?
> I would like to know why aplay -l still show "no soundcard" after ./sof_insert.sh.
> 
> Thanks

@bardliao i  tried 10 times manually
after step 2:
 1st ,2nd ,7th, 9th aplay -l  can see pcm list
 3rd, 4th, 5th, 6th ,8th, 10th aplay -l shows "no soundcard"

add dmesg below
[normal-dmesg.log](https://github.com/thesofproject/linux/files/3771346/normal-dmesg.log)
[nosoundcard-dmesg.log](https://github.com/thesofproject/linux/files/3771347/nosoundcard-dmesg.log)

**Comment by bardliao** [comment]
@dengyangchao I can see **mapping ok** from the nosoundcard-dmesg.log. I am wondering did you check aplay -l too early? Please note that the sound card will not be created immediately. Please check it at least after 10 sec.

```
[    4.112285] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #16 (SDW0-Playback)
[    4.112286] rt711_rt1308 rt711_rt1308: rt711-aif1 <-> SDW0 Pin2 mapping ok
[    4.112287] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #17 (SDW0-Capture)
[    4.112288] rt711_rt1308 rt711_rt1308: rt711-aif1 <-> SDW0 Pin3 mapping ok
[    4.112291] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #18 (SSP2-Codec)
[    4.112291] rt711_rt1308 rt711_rt1308: rt1308-aif <-> SSP2 Pin mapping ok
[    4.112292] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #19 (dmic01)
[    4.112293] rt711_rt1308 rt711_rt1308: dmic-hifi <-> DMIC01 Pin mapping ok
[    4.112294] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #20 (dmic16k)
[    4.112295] rt711_rt1308 rt711_rt1308: dmic-hifi <-> DMIC16k Pin mapping ok
[    4.112300] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #0 SDW0-headphone (*)
[    4.112302] sof-audio-pci 0000:00:1f.3: creating new PCM SDW0-headphone
[    4.112303] sof-audio-pci 0000:00:1f.3: spcm: allocate Passthrough Playback 0 playback DMA buffer size 0x10000 max 0x10000
[    4.112309] rt711_rt1308 rt711_rt1308: snd-soc-dummy-dai <-> SDW0-headphone 0 mapping ok
[    4.112313] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #1 SDW0-mics (*)
[    4.112314] sof-audio-pci 0000:00:1f.3: creating new PCM SDW0-mics
[    4.112314] sof-audio-pci 0000:00:1f.3: spcm: allocate Passthrough Capture 1 capture DMA buffer size 0x10000 max 0x10000
[    4.112319] rt711_rt1308 rt711_rt1308: snd-soc-dummy-dai <-> SDW0-mics 1 mapping ok
[    4.112322] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #2 Speaker (*)
[    4.112322] sof-audio-pci 0000:00:1f.3: creating new PCM Speaker
[    4.112323] sof-audio-pci 0000:00:1f.3: spcm: allocate Passthrough Playback 2 playback DMA buffer size 0x10000 max 0x10000
[    4.112328] rt711_rt1308 rt711_rt1308: snd-soc-dummy-dai <-> Speaker 2 mapping ok
[    4.112332] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #3 DMIC01 (*)
[    4.112332] sof-audio-pci 0000:00:1f.3: creating new PCM DMIC01
[    4.112333] sof-audio-pci 0000:00:1f.3: spcm: allocate Passthrough Capture 3 capture DMA buffer size 0x10000 max 0x10000
[    4.112338] rt711_rt1308 rt711_rt1308: snd-soc-dummy-dai <-> DMIC01 3 mapping ok
[    4.112349] rt711_rt1308 rt711_rt1308: ASoC: registered pcm #4 DMIC16k (*)
[    4.112351] sof-audio-pci 0000:00:1f.3: creating new PCM DMIC16k
[    4.112353] sof-audio-pci 0000:00:1f.3: spcm: allocate Passthrough Capture 4 capture DMA buffer size 0x10000 max 0x10000
[    4.112365] rt711_rt1308 rt711_rt1308: snd-soc-dummy-dai <-> DMIC16k 4 mapping ok
```

**Comment by dengyangchao** [comment]
@bardliao i tried another time, and wait for a long time, aplay -l still "no soundcard"

<details>
 <summary>Dmesg</summary>

````
[   43.527229] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x40020000: GLB_PM_MSG: CTX_RESTORE
[   43.527236] intel-sdw sdw-master-0: intel_resume_runtime start
[   43.527423] rt711 sdw:0:25d:711:0: sdw_modify_slave_status: initializing completion for Slave 1
[   43.527541] intel-sdw sdw-master-0: intel_resume_runtime done
[   43.528819] intel-sdw sdw-master-0: Slave status change
[   43.528828] intel-sdw sdw-master-0: sdw_handle_slave_status: start
[   43.528832] intel-sdw sdw-master-0: Slave attached, programming device number
[   43.529012] intel-sdw sdw-master-0: SDW Slave Addr: 20025d071100
[   43.529015] intel-sdw sdw-master-0: SDW Slave class_id 0, part_id 711, mfg_id 25d, unique_id 0, version 2
[   43.529018] intel-sdw sdw-master-0: in sdw_assign_device_num
[   43.529020] intel-sdw sdw-master-0: in sdw_assign_device_num
[   43.529022] intel-sdw sdw-master-0: Slave already registered, reusing dev_num:1
[   43.529024] intel-sdw sdw-master-0: in sdw_assign_device_num, writing dev_num 1
[   43.529291] intel-sdw sdw-master-0: Msg Ack not received
[   43.529294] intel-sdw sdw-master-0: Msg Ack not received
[   43.529296] intel-sdw sdw-master-0: Msg Ack not received
[   43.529298] intel-sdw sdw-master-0: Msg Ack not received
[   43.529300] intel-sdw sdw-master-0: Msg Ack not received
[   43.529302] intel-sdw sdw-master-0: Msg Ack not received
[   43.529311] intel-sdw sdw-master-0: Msg ignored for Slave 0
[   43.529314] intel-sdw sdw-master-0: No more devices to enumerate
[   43.529316] intel-sdw sdw-master-0: End of sdw_program_device_num ret 0
[   43.529354] intel-sdw sdw-master-0: Slave status change
[   43.529375] intel-sdw sdw-master-0: sdw_handle_slave_status: start
[   43.529379] intel-sdw sdw-master-0: processing Slave 1 Status 1
[   43.529382] rt711 sdw:0:25d:711:0: sdw_modify_slave_status: signaling completion for Slave 1
[   43.529386] intel-sdw sdw-master-0: Slave 1 initializing
[   43.529483] intel-sdw sdw-master-0: intel_startup: SDW0 Pin3: start
[   43.529505] intel-sdw sdw-master-0: intel_startup: SDW0 Pin3: done
[   43.529513]  SDW0-mics: ASoC: open FE SDW0-mics
[   43.529521] sof-audio-pci 0000:00:1f.3: pcm: open stream 1 dir 1
[   43.529524] sof-audio-pci 0000:00:1f.3: period min 192 max 16384 bytes
[   43.529525] sof-audio-pci 0000:00:1f.3: period count 2 max 16
[   43.529527] sof-audio-pci 0000:00:1f.3: buffer max 65536 bytes
[   43.529547] intel-sdw sdw-master-0: Updating Slave 1 status
[   43.529556] intel-sdw sdw-master-0: sdw_handle_slave_status: Updating Slave 1 status done
[   43.529577] intel-sdw sdw-master-0: sdw_handle_slave_status: end
[   43.529909]  SDW0-Capture: ASoC: hw_params BE SDW0-Capture
[   43.529916] rt711 sdw:0:25d:711:0: rt711_pcm_hw_params rt711-aif1
[   43.530092] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 7203 8283 <= 0041
[   43.530228] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 7209 8289 <= 0041
[   43.530383] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 7208 8288 <= 0041
[   43.530385] rt711 sdw:0:25d:711:0: Update DAI routes for rt711-aif1 capture
[   43.530389] rt711 sdw:0:25d:711:0: Connecting DAI route DP4TX -> DP4 Capture
[   43.530392] intel-sdw sdw-master-0: intel_hw_params: SDW0 Pin3: start
[   43.530411] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[   43.530487] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[   43.530489] intel-sdw sdw-master-0: intel_hw_params: SDW0 Pin3: done
[   43.530494] intel-sdw sdw-master-0: Update DAI routes for SDW0 Pin3 capture
[   43.530497]  SDW0-mics: ASoC: hw_params FE SDW0-mics rate 48000 chan 2 fmt 2
[   43.530499] sof-audio-pci 0000:00:1f.3: Update DAI routes for SDW0-mics 1 capture
[   43.530501] sof-audio-pci 0000:00:1f.3: Connecting DAI route PCM1C -> Passthrough Capture 1
[   43.530503] sof-audio-pci 0000:00:1f.3: pcm: hw params stream 1 dir 1
[   43.530507] sof-audio-pci 0000:00:1f.3: generating page table for 00000000a4d7bfd0 size 0xffc0 pages 16
[   43.530520] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
[   43.530555] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
[   43.530558] sof-audio-pci 0000:00:1f.3: period_bytes:0x3fc0
[   43.530570] sof-audio-pci 0000:00:1f.3: periods:4
[   43.530586] sof-audio-pci 0000:00:1f.3: stream_tag 2
[   43.530594] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[   43.530754] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[   43.530757] sof-audio-pci 0000:00:1f.3: pcm: stream dir 1, posn mailbox offset is 790528
[   43.530782]  SDW0-mics: ASoC: prepare FE SDW0-mics
[   43.530784]  SDW0-Capture: ASoC: prepare BE SDW0-Capture
[   43.530787] intel-sdw sdw-master-0: intel_prepare: SDW0 Pin3: start
[   43.530815] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[   43.530920] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[   43.530922] sdw_prepare_stream: SDW0 Pin3-Capture start
[   43.531762] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00e0 <= 0003
[   43.531838] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00f0 <= 0003
[   43.531841] rt711 sdw:0:25d:711:0: rt711_clock_config complete, clk_freq=4800000
[   43.532020] sdw_prepare_stream: SDW0 Pin3-Capture: done
[   43.532030] intel-sdw sdw-master-0: intel_prepare: SDW0 Pin3: done
[   43.532304] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 3501 <= 0003
[   43.532408] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 3501 <= 0000
[   43.532508] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 3608 <= 0010
[   43.532537]  SDW0-Capture: ASoC: BE SDW0-Capture event 1 dir 1
[   43.533030]  SDW0-Capture: ASoC: trigger BE SDW0-Capture cmd 1
[   43.533034] intel-sdw sdw-master-0: intel_trigger: SDW0 Pin3: start
[   43.533039] sdw_enable_stream: SDW0 Pin3-Capture: start
[   43.533838] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00e0 <= 0003
[   43.533877] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00f0 <= 0003
[   43.533878] rt711 sdw:0:25d:711:0: rt711_clock_config complete, clk_freq=4800000
[   43.534002] sdw_enable_stream: SDW0 Pin3-Capture: done
[   43.534004] intel-sdw sdw-master-0: intel_trigger: SDW0 Pin3: done
[   43.534006]  SDW0-mics: ASoC: post trigger FE SDW0-mics cmd 1
[   43.534007] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 1 dir 1 cmd 1
[   43.534529] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x2024001e successful
[   43.534531] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60040000: GLB_STREAM_MSG: TRIG_START
[   43.534678] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60040000: GLB_STREAM_MSG: TRIG_START
[   44.026803] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[   44.026827] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[   44.526976] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[   44.526997] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[   46.177188]  SDW0-mics: ASoC: pre trigger FE SDW0-mics cmd 0
[   46.177195] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 1 dir 1 cmd 0
[   46.177200] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[   46.177483] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[   46.178016] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x240000 successful
[   46.178033] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[   46.178238] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[   46.178246]  SDW0-Capture: ASoC: trigger BE SDW0-Capture cmd 0
[   46.178252] intel-sdw sdw-master-0: intel_trigger: SDW0 Pin3: start
[   46.178262] sdw_disable_stream: SDW0 Pin3-Capture: start
[   46.179515] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00e0 <= 0003
[   46.179617] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00f0 <= 0003
[   46.179621] rt711 sdw:0:25d:711:0: rt711_clock_config complete, clk_freq=4800000
[   46.180213] sdw_disable_stream: SDW0 Pin3-Capture: done
[   46.180223] intel-sdw sdw-master-0: intel_trigger: SDW0 Pin3: done
[   46.180243]  SDW0-mics: ASoC: hw_free FE SDW0-mics
[   46.180248] sof-audio-pci 0000:00:1f.3: pcm: free stream 1 dir 1
[   46.180264]  SDW0-Capture: ASoC: hw_free BE SDW0-Capture
[   46.180268] intel-sdw sdw-master-0: intel_hw_free: SDW0 Pin3: start
[   46.180272] sdw_deprepare_stream: SDW0 Pin3-Capture: start
[   46.180525] sdw_deprepare_stream: SDW0 Pin3-Capture: done
[   46.180541] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[   46.180724] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[   46.180740] intel-sdw sdw-master-0: intel_hw_free: SDW0 Pin3: done
[   46.180889]  SDW0-Capture: ASoC: close BE SDW0-Capture
[   46.180902] intel-sdw sdw-master-0: intel_shutdown: SDW0 Pin3: start
[   46.180928] intel-sdw sdw-master-0: intel_shutdown: SDW0 Pin3: done
[   46.180974]  SDW0-mics: ASoC: close FE SDW0-mics
[   46.180982] sof-audio-pci 0000:00:1f.3: pcm: close stream 1 dir 1
[   46.181175] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 3608 <= 0000
[   46.181434] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 3501 <= 0003
[   46.181465]  SDW0-Capture: ASoC: BE SDW0-Capture event 2 dir 1
[   46.181475]  SDW0-mics: ASoC: BE capture disconnect check for SDW0-Capture
[   46.181477]  SDW0-mics: freed DSP capture path SDW0-mics <- SDW0-Capture
[   47.026869] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[   47.026890] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[   49.321723] intel-sdw sdw-master-0: intel_suspend_runtime start
[   49.321797] intel-sdw sdw-master-0: intel_suspend_runtime done
[   49.321805] sof-audio-pci 0000:00:1f.3: snd_sof_runtime_idle
[   49.321811] sof-audio-pci 0000:00:1f.3: snd_sof_runtime_suspend
[   49.321814] sof-audio-pci 0000:00:1f.3: sof_suspend
[   49.321828] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x20140000 successful
[   49.347163] sof-audio-pci 0000:00:1f.3: sof_send_pm_ipc
[   49.347170] sof-audio-pci 0000:00:1f.3: ipc tx: 0x40010000: GLB_PM_MSG: CTX_SAVE
[   49.347471] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x40010000: GLB_PM_MSG: CTX_SAVE
[   49.347491] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf010f0f successful
[   49.347500] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf000f0f successful
[   49.347505] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 0 : core_mask 1
[   49.347585] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000010 at  00000044
````
</details>




[dmesg.log](https://github.com/thesofproject/linux/files/3771738/dmesg.log)

**Comment by bardliao** [comment]
@dengyangchao Could you double check if you get the right dmesg.log? dmesg.log and nosoundcard-dmesg.log are exactly the same (even the timestamp are the same)?

**Comment by dengyangchao** [comment]
> @dengyangchao Could you double check if you get the right dmesg.log? dmesg.log and nosoundcard-dmesg.log are exactly the same (even the timestamp are the same)?

i think it is the right one, i save it as different name.  And dose the info intact?  i mean after the wait 10s. Because i tried the same commit more than10 times today (after 2 days power off). run ./sof_insert.sh , "no soundcard" not occur

**Comment by RanderWang** [comment]
this bug is very odd,  EDP affect audio codec

**Comment by RanderWang** [comment]
without EDP display plugged,  headphone jack detection is always timeout.

**Comment by plbossart** [comment]
can you try on ICL with the 3-in-1 board? if this works then you have a TGL-specific issue. If not, then it's an RT711 issue. It's impossible to progress without more information on this bug.
Also take HDMI out completely, there is no reason why HDMI/DP should interfere.

**Comment by YvonneYang2** [comment]
Check it on icl 3-in-1 board with kernel removed HDMI parts and no-hdmi tplg.
Original issue can be reproduced, arecord always capture no audio, same as  https://github.com/thesofproject/linux/issues/1333#issuecomment-543104300 .But difference from original issue is that aplay0,0 always has sound output.

Arecord Logs:
[dmesg.log](https://github.com/thesofproject/linux/files/3783159/dmesg.log)
[error.log](https://github.com/thesofproject/linux/files/3783160/error.log)
[logger.log](https://github.com/thesofproject/linux/files/3783161/logger.log)

Note:
I tired that firstly set rt711_amixer after system boot up with headset connected,  then check headset jack and headset mic jack amixer value, they turned on and arecord and aplay can work normally. 

rt711_amixer setting:
[rt711_mixer.sh.txt](https://github.com/thesofproject/linux/files/3783179/rt711_mixer.sh.txt)

**Comment by sinahuang** [comment]
Original issue can not be reproduced on TGL RVP with codec ACL711, ALC1308 in SDW mode since PR #1392 merged into soundwire-latest.

Test recipe:
Kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 9f1bd5b
FW: https://github.com/thesofproject/sof/commits/master commit: 2ca6a4a
tplg: Same with TGL FW branch commit: sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711-rt1308.tplg

**Comment by YvonneYang2** [comment]
Issue still can be reproduced on ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode.

Steps:
1. Boot up with headset connected.
2. Check headphone jack and headset mic jack vlaues --> off.
3. Aplay0,0 via " aplay -Dhw:0,0 -r48000 -c2 -fs16_le test.wav -vv -i " --> headset has sound output.
4. Check headphone jack and headset mic jack values --> on.
5. Open sound setting , then unplug headset. --> headphone jack and headset mic jack value shows on, headphone option is still in sound setting.
6. Plug in headset again --> headphone option disappeared in sound setting, headphone jack and headset jack values are off.  Jack status is opposite of headset hot-plug action.

Test recipe:
Kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 9f1bd5b
FW: https://github.com/thesofproject/sof/commits/master commit: 2ca6a4a
tplg: Same with FW branch, sof-icl-rt711-rt1308-rt715-hdmi.tplg  rename  to  sof-icl-rt711-rt1308-rt715.tplg
platform: ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode

[dmesg.log](https://github.com/thesofproject/linux/files/3788133/dmesg.log)
[logger.log](https://github.com/thesofproject/linux/files/3788134/logger.log)

**Comment by plbossart** [comment]
@YvonneYang2 the code to handle the Rt711 is identical in the two machine drivers. The codec driver is also identical.

Also what are you testing? The bug title is 'Can't hear audio output when aplay via headset in first time', and your own write up says this is solved.

so can you talk with @sinahuang can double check that you report results on the same problem....

**Comment by YvonneYang2** [comment]
Changed rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c , then tested it again, issue cannot be reproduced on ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode. 

Test recipe:
kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 9f1bd5b 
FW: https://github.com/thesofproject/sof/commits/master  commit: 59c9338
tplg: Same with FW branch, sof-icl-rt711-rt1308-rt715-hdmi.tplg  rename  to  sof-icl-rt711-rt1308-rt715.tplg
platform: ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode

**Comment by plbossart** [comment]
@YvonneYang2 do you mean you modified code or you updated the settings with alsamixer. please clarify?

**Comment by YvonneYang2** [comment]
@plbossart 
Issue cannot be reproduced if the code is modified(Change rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c). No need to update the setting with alsamixer.

**Comment by bardliao** [comment]
> @plbossart
> Issue cannot be reproduced if the code is modified(Change rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c). No need to update the setting with alsamixer.

@YvonneYang2 I believe that we could change it by ACPI since codec driver gets jd_src by device_property_read_u32().  @plbossart Do you know how to set device property for sdw devices on ACPI?

**Comment by YvonneYang2** [comment]
JD source issue will track in #1420
Since this issue not reproduced, close it.

**Comment by plbossart** [comment]
> > @plbossart
> > Issue cannot be reproduced if the code is modified(Change rt711->jd_src from RT711_JD2 to RT711_JD1 in rt711.c). No need to update the setting with alsamixer.
> 
> @YvonneYang2 I believe that we could change it by ACPI since codec driver gets jd_src by device_property_read_u32(). @plbossart Do you know how to set device property for sdw devices on ACPI?

This is what is used in existing machine drivers (bytcr_rt5640)

````c
static int byt_rt5640_add_codec_device_props(const char *i2c_dev_name)
{
	struct property_entry props[MAX_NO_PROPS] = {};
	struct device *i2c_dev;
	int ret, cnt = 0;

	i2c_dev = bus_find_device_by_name(&i2c_bus_type, NULL, i2c_dev_name);
	if (!i2c_dev)
		return -EPROBE_DEFER;

	switch (BYT_RT5640_MAP(byt_rt5640_quirk)) {
	case BYT_RT5640_DMIC1_MAP:
		props[cnt++] = PROPERTY_ENTRY_U32("realtek,dmic1-data-pin",
						  RT5640_DMIC1_DATA_PIN_IN1P);
		break;
	case BYT_RT5640_DMIC2_MAP:
		props[cnt++] = PROPERTY_ENTRY_U32("realtek,dmic2-data-pin",
						  RT5640_DMIC2_DATA_PIN_IN1N);
		break;
	case BYT_RT5640_IN1_MAP:
		if (byt_rt5640_quirk & BYT_RT5640_DIFF_MIC)
			props[cnt++] =
				PROPERTY_ENTRY_BOOL("realtek,in1-differential");
		break;
	case BYT_RT5640_IN3_MAP:
		if (byt_rt5640_quirk & BYT_RT5640_DIFF_MIC)
			props[cnt++] =
				PROPERTY_ENTRY_BOOL("realtek,in3-differential");
		break;
	}

	if (BYT_RT5640_JDSRC(byt_rt5640_quirk)) {
		props[cnt++] = PROPERTY_ENTRY_U32(
				    "realtek,jack-detect-source",
				    BYT_RT5640_JDSRC(byt_rt5640_quirk));

		props[cnt++] = PROPERTY_ENTRY_U32(
				    "realtek,over-current-threshold-microamp",
				    BYT_RT5640_OVCD_TH(byt_rt5640_quirk) * 100);

		props[cnt++] = PROPERTY_ENTRY_U32(
				    "realtek,over-current-scale-factor",
				    BYT_RT5640_OVCD_SF(byt_rt5640_quirk));
	}

	if (byt_rt5640_quirk & BYT_RT5640_JD_NOT_INV)
		props[cnt++] = PROPERTY_ENTRY_BOOL("realtek,jack-detect-not-inverted");

	ret = device_add_properties(i2c_dev, props);
	put_device(i2c_dev);

	return ret;
}
````

I think the first function should just work if we use a &sdw_bus_type?

**Comment by sinahuang** [comment]
jhklj
